internal/handlers/createlink.go

```diff
@@ -37,14 +62,31 @@ func (s *Server) CreateLinkJSON(writer http.ResponseWriter, request *http.Reques
 	}
 
 	key := s.generator.CreateKey()
-	s.storage.Add(key, url)
-
-	resp := models.ResponseCreateLink{
-		Result: fmt.Sprintf("%s/%s", s.Config.ResolveHost, key),
+	err := s.storage.Add(key, url)
+	if err != nil {
+		if err == storage.ErrConflict {
+			shortURL, err := s.storage.GetByOriginURL(url)
+			if err != nil {
+				http.Error(writer, "Internal Backend Error", http.StatusInternalServerError)
+				return
+			}
+			respStatus = http.StatusConflict
+			resp = models.ResponseCreateLink{
+				Result: fmt.Sprintf("%s/%s", s.Config.ResolveHost, shortURL),
+			}
```

Comment:
если сразу возвращать ответ, то не придется делать бесконечные else {
```suggestion
writer.WriteHeader(respStatus)
return
			}
```

Reply:
Вообще да, но тогда придётся дублировать 3 строчки кода

```
	writer.Header().Set("Content-Type", "application/json")
	writer.WriteHeader(respStatus)

	encoder := json.NewEncoder(writer)
```